packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -386,3 +544,689 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
+      GestureRecognizerFactoryWithHandlers<_ThumbPressGestureRecognizer>(
+          () => _ThumbPressGestureRecognizer(
+          debugOwner: this,
+          customPaintKey: _scrollbarPainterKey,
+          pressDuration: widget.pressDuration,
+        ),
+          (_ThumbPressGestureRecognizer instance) {
+          instance
+            ..onLongPress = handleThumbPress
+            ..onLongPressStart = handleThumbPressStart
+            ..onLongPressMoveUpdate = handleThumbPressUpdate
+            ..onLongPressEnd = handleThumbPressEnd;
+        },
+      );
+
+    gestures[_TrackTapGestureRecognizer] =
+      GestureRecognizerFactoryWithHandlers<_TrackTapGestureRecognizer>(
+          () => _TrackTapGestureRecognizer(
+          debugOwner: this,
+          customPaintKey: _scrollbarPainterKey,
+        ),
+          (_TrackTapGestureRecognizer instance) {
+          instance.onTapDown = _handleTrackTapDown;
+        },
+      );
+
+    return gestures;
+  }
+  /// Returns true if the provided [Offset] is located over the track of the
+  /// [RawScrollbar].
+  ///
+  /// Excludes the [RawScrollbar] thumb.
+  @protected
+  bool isPointerOverTrack(Offset position) {
+    if (_scrollbarPainterKey.currentContext == null) {
+      return false;
+    }
+    final Offset localOffset = _getLocalOffset(_scrollbarPainterKey, position);
+    return scrollbarPainter.hitTestInteractive(localOffset)
+      && !scrollbarPainter.hitTestOnlyThumbInteractive(localOffset);
+  }
+  /// Returns true if the provided [Offset] is located over the thumb of the
+  /// [RawScrollbar].
+  @protected
+  bool isPointerOverThumb(Offset position) {
+    if (_scrollbarPainterKey.currentContext == null) {
+      return false;
+    }
+    final Offset localOffset = _getLocalOffset(_scrollbarPainterKey, position);
+    return scrollbarPainter.hitTestOnlyThumbInteractive(localOffset);
+  }
+  /// Returns true if the provided [Offset] is located over the track or thumb
+  /// of the [RawScrollbar].
+  @protected
+  bool isPointerOverScrollbar(Offset position) {
+    if (_scrollbarPainterKey.currentContext == null) {
+      return false;
+    }
+    final Offset localOffset = _getLocalOffset(_scrollbarPainterKey, position);
+    return scrollbarPainter.hitTestInteractive(localOffset);
+  }
+
+  /// Cancels the fade out animation so the scrollbar will remain visible for
+  /// interaction.
+  ///
+  /// Can be overridden by subclasses to respond to a [PointerHoverEvent].
+  ///
+  /// Helper methods [isPointerOverScrollbar], [isPointerOverThumb], and
+  /// [isPointerOverTrack] can be used to determine the location of the pointer
+  /// relative to the painter scrollbar elements.
+  @protected
+  @mustCallSuper
+  void handleHover(PointerHoverEvent event) {
+    // Check if the position of the pointer falls over the painted scrollbar
+    if (isPointerOverScrollbar(event.position)) {
+      _hoverIsActive = true;
+      _fadeoutTimer?.cancel();
```

Comment:
This can easily cause the thumb to remain on the touch device after release the flinger when `ScrollBar` nested.
because the `Tap` event will also trigger a `Hover` event that cancels the dismiss timer and no change to restart the fadeout timer until the next touch event dispatch.
![20201208_192438](https://user-images.githubusercontent.com/61075224/101479363-0ad78300-398d-11eb-91a0-ca215f7c51ac.gif)

Reply:
### Sample Code
```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(SampleWidget());
}

class SampleWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Scrollbar(
//          thickness: 30,
          child: SingleChildScrollView(
            child: SizedBox(
              height: 1000.0,
              width: double.infinity,
              child: Column(
                children: [
                  Scrollbar(
//                    thickness: 15,
                    child: SizedBox(
                      height: 500.0,
                      width: double.infinity,
                      child: SingleChildScrollView(
                        child: SizedBox(
                          key: Key('Inner scrollable'),
                          height: 1000.0,
                          child: ColoredBox(
                            color: Colors.pink,
                            child: FlutterLogo(),
                          ),
                        ),
                      ),
                    ),
                  ),
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }
}

```

Reply:
There is no problem in non-nested scenarios because the `handleThumbPressEnd` will restart the timer, but when nested the `pressEnd` event was caught by the outer scrollbar.

Reply:
See below comments. I am not sure I fully understand the expected behaviors when stacks Scrollbars on top of one another.